### ISSUE TITLE
UpdateRequestBuilder: Edm.Date as Key of Entity ends in exception "value.format is not a function" 

### ISSUE THREAD

**Issue by drvup** [SAP/cloud-sdk-js#3841]
**Describe the bug**
Hi Cloud SDK team for ts/js,
I am implementing an integration scenario where I am using the SDK, like always ;-), to create an update request for a service, hosted on a CAP app. I have a generated VDM in use, to interact elegant with the service.

Within my entity - a general ledger report -  I have multiple key fields with different types behind:
Edm.String: companyCode, context_region, ledgerType, reportType
Edm.Date: periodEndDate

Each update request I do fire results within an error message:
"value.format is not a function".
This comes from `@sap-cloud-sdk/odata-v4/dist/de-serializers/converters.js > serializeToDate()`, what is the "defaultSerializer" for Edm.Date type. So I checked `periodEndDate` contains as value, a `moment` object. And this I can confirm.

So I did debug the compiled code from you guys and did identify, it's related to the fact that `periodEndDate` is part of the key of my entity.
For keys of an entity, you're going to encode the value (as it's part of the URL, it does make sense to do so) of each key within function `encodeURIComponent`. This is happening here:
![encodeUri](https://user-images.githubusercontent.com/9018930/236146433-2a1a5844-e90b-4bb6-8d6c-371854867ee2.png)

The result is visible within the `UpdateRequestBuilderBase`:
![updateRequestBuilderClass](https://user-images.githubusercontent.com/9018930/236146630-ac240f92-744d-4fc0-809b-b3bac96a8f0f.png)
You can see, the encoded value of the `moment`-Object of `periodEndDate`.

So it absolutely does make sense, the "defaultSerializer" can't proceed further as there is no `moment`-Object existing:
![uncaughtException](https://user-images.githubusercontent.com/9018930/236147226-0bf01db6-0e4e-4848-8f9a-b9259b0254a0.png)

=> result: "value.format is not a function"

**To Reproduce**
Set up a oData service within a VDM, having a `Edm.Date` as key. Example CDS:
Entity:

```
/**
 * GlAccountStandardReport Report entity
 */
entity GlAccountStandardReport : cuid, managed {
  reportType                       : String(50) not null;
  periodEndDate                    : Date not null;
  companyCode                      : String(50) not null;
  ledgerType                       : String(50) not null;
  erpCreateDateTime                : Timestamp;
  erpChangeDateTime                : Timestamp;
}
```

and the service....
```
service IngestionLayerService @(requires : ['test-user']) {

    entity GlAccountStandardReport @(restrict: [
        { grant: '*', to: 'test-user' }
        ]) as projection on GlAccountStandardReport{*,
        ID @Core.Computed,
        key context.region as context_region,
        key companyCode,
        key reportType,
        key periodEndDate,
        key ledgerType,
        } excluding{context, localized};
```

**Expected behavior**
The SDK needs to handle the `moment` object as part of the key in another approach as within normal strings, as the `serializers` of the "defaultSerializers" of type `Edm.Date` needs to be able to handle it. 

**Used Versions:**
- node version: v18.15.0
- npm version: v9.5.0
- SAP Cloud SDK: 3.1.1
- SAP Cloud SDK Generator (of created and consumed Virtual Data Model): 3.1.1

**Impact / Priority**
Affected development phase: Development (4 weeks until SIT)
Impact: Blocked (as `periodEndDate` needs to remain a key-field) *

(*) we're checking the ability to change the type from `Edm.Date` to `Edm.String` to mitigate the issue.

**Additional context**
Within our environment, a CAP application represents some kind of unified-data-store, providing us multiple services / oData endpoints. Each service gets automatically VDM's generated (as soon as an update got detected) and published to a private Azure DevOps Artifact Feed. Other applications do use these generated VDMs to consume / update data from / to the CAP app.

Sorry for the long description, but I really hope more context and details is helping you to negotiate that blocker issue.

Thanks and kind regards,
Cedric

**Comment by deekshas8** [comment]
Hi @drvup ,

Thanks for raising this and your detailed investigation, this is indeed a bug. I have created an issue for this in our backlog. Will update you when we have a fix.

**Comment by deekshas8** [comment]
Hi @drvup ,

We fixed the issue in the latest v3.2.0 release. Please try it out.
